src/ioctls/vm.rs

```diff
@@ -1358,6 +1426,68 @@ mod tests {
         assert!(vm_fd.unregister_irqfd(&evtfd3, 5).is_ok());
     }
 
+    #[test]
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    fn test_set_irq_line() {
+        let kvm = Kvm::new().unwrap();
+        let vm_fd = kvm.create_vm().unwrap();
+
+        assert!(vm_fd.create_irq_chip().is_ok());
+
+        assert!(vm_fd.set_irq_line(4, true).is_ok());
+        assert!(vm_fd.set_irq_line(4, false).is_ok());
+        assert!(vm_fd.set_irq_line(4, true).is_ok());
+    }
+
+    #[test]
+    #[cfg(target_arch = "aarch64")]
+    fn test_set_irq_line() {
+        let kvm = Kvm::new().unwrap();
+        let vm_fd = kvm.create_vm().unwrap();
+        // Create a vcpu for test case 2 of the KVM_IRQ_LINE API on aarch64.
```

Comment:
What do you mean with test case 2 here?

Reply:
sure.

In below comments:
```
// On arm/aarch64, irq field is interpreted like this:
// bits:  | 31 ... 24 | 23  ... 16 | 15    ...    0 |
// field: | irq_type  | vcpu_index |     irq_id     |
// The irq_type field has the following values:
// - irq_type[0]: out-of-kernel GIC: irq_id 0 is IRQ, irq_id 1 is FIQ
// - irq_type[1]: in-kernel GIC: SPI, irq_id between 32 and 1019 (incl.) (the vcpu_index field is ignored)
// - irq_type[2]: in-kernel GIC: PPI, irq_id between 16 and 31 (incl.)
// Hence, using irq_type = 1, irq_id = 32 (decimal), the irq field in hex is: 0x01_00_0020
// Case 2: using irq_type = 2, vcpu_index = 0, irq_id = 16 (decimal), the irq field in hex is: 0x02_00_0010
```

 `irq_type` can be either `1` or `2`, I am using case 1, 2 as the cases for testing, namely `irq_type` equals `1` or `2`. Therefore case 2 means using `irq_type = 2`, `vcpu_index = 0`, `irq_id = 16 (decimal)`, the irq field in hex is then: `0x02_00_0010`